### PR TITLE
Ind support update

### PR DESCRIPTION
**Pull request by ZhangMan1126**
fix and update for meituan case studies

### CODE REVIEW

File: client-java/controller/src/main/java/org/evomaster/client/java/controller/internal/EMController.java [hunk 1]
```diff
@@ -223,11 +223,24 @@ public Response getSutInfo(@Context HttpServletRequest httpServletRequest) {
...
+                try{
+                    // handled seeded tests
+                    dto.rpcProblem.seededTestDtos = noKillSwitch(() -> sutController.handleSeededTests());
+
+                    if (dto.isSutRunning){
+                        noKillSwitch(() -> sutController.getSeededExternalServiceResponseDto());
+                    }
+                }catch (RuntimeException e){
+                    StringBuilder msg = new StringBuilder("Fail to handle specified seeded tests " + e.getMessage());
+                    if (e.getStackTrace() != null && e.getStackTrace().length > 0){
```
Comment (arcuri82) [inline client-java/controller/src/main/java/org/evomaster/client/java/controller/internal/EMController.java:236]: you are calling `getStackTrace()` 4 times. note that each call make a copy... maybe best to call it once, and save it in variable

File: client-java/controller/src/main/java/org/evomaster/client/java/controller/problem/rpc/CodeJavaGenerator.java [hunk 2]
```diff
@@ -303,6 +303,15 @@ public static String junitAssertEquals(String expectedValue, String variableName
         if (AssertionsUtil.getAssertionsWithComment(assertionScript)){
             return "//" + assertionScript;
         }
+
+        /*
+            if length of expected value is more than 10, comment out this assertion for the moment
+
+            TODO need to better handle flakiness later
+         */
+        if (expectedValue.length() > 10)
```
Comment (arcuri82) [inline client-java/controller/src/main/java/org/evomaster/client/java/controller/problem/rpc/CodeJavaGenerator.java:312]: hmmmm. not sure i like this. but i guess it is fine for the moment as it applied only for RPC. but please create a Trello card mentioning this file and TODO

File: client-java/controller/src/main/java/org/evomaster/client/java/controller/problem/rpc/RPCExceptionHandler.java [hunk 3]
```diff
@@ -125,11 +125,11 @@ private static boolean handleThrift(Object e, EndpointSchema endpointSchema, RPC
             handled = handleTException(e, dto);
 
             if (!handled){
-                SimpleLogger.error("Fail to extract exception type info for an exception "+ e.getClass().getName());
+                SimpleLogger.uniqueWarn("Fail to extract exception type info for an exception "+ e.getClass().getName());
             }
 
         } catch (ClassNotFoundException ex) {
-            SimpleLogger.error("ERROR: in handling Thrift exception with error msg:"+ex.getMessage());
+            SimpleLogger.uniqueWarn("ERROR: in handling Thrift exception with error msg:"+ex.getMessage());
```
Comment (arcuri82) [inline client-java/controller/src/main/java/org/evomaster/client/java/controller/problem/rpc/RPCExceptionHandler.java:132]: are those `errors` (faults in EM) or `warnings` (other kinds of issues that are not critical)?  it is weird to see a `ERROR: ...` txt in a `warn` message

File: client-java/controller/src/main/java/org/evomaster/client/java/controller/problem/rpc/RPCExceptionHandler.java [hunk 4]
```diff
@@ -179,7 +183,7 @@ private static String getExceptionMessage(Object e)  {
             getMessage.setAccessible(true);
             return (String) getMessage.invoke(e);
         } catch (NoSuchMethodException | InvocationTargetException | IllegalAccessException ex) {
-            SimpleLogger.error("Error: fail to get message of the exception with "+ex.getMessage());
+            SimpleLogger.uniqueWarn("Error: fail to get message of the exception with "+ex.getMessage());
```
Comment (arcuri82) [inline client-java/controller/src/main/java/org/evomaster/client/java/controller/problem/rpc/RPCExceptionHandler.java:186]: see previous comments

File: client-java/controller/src/main/java/org/evomaster/client/java/controller/problem/rpc/RPCExceptionHandler.java [hunk 5]
```diff
@@ -197,7 +201,7 @@ private static Object getExceptionCause(Object e)  {
             getCause.setAccessible(true);
             return getCause.invoke(e);
         } catch (NoSuchMethodException | InvocationTargetException | IllegalAccessException ex) {
-            SimpleLogger.error("Error: fail to get message of the exception with "+ex.getMessage());
+            SimpleLogger.uniqueWarn("Error: fail to get message of the exception with "+ex.getMessage());
```
Comment (arcuri82) [inline client-java/controller/src/main/java/org/evomaster/client/java/controller/problem/rpc/RPCExceptionHandler.java:204]: see previous comments

**Review by arcuri82 — CHANGES_REQUESTED** [review]

File: client-java/controller/src/main/java/org/evomaster/client/java/controller/problem/rpc/CodeJavaGenerator.java (hunk 2 above)
Comment (ZhangMan1126) [reply client-java/controller/src/main/java/org/evomaster/client/java/controller/problem/rpc/CodeJavaGenerator.java:312]: this now is clarified in this card, https://trello.com/c/v5XRNyml/707-flakiness-configurable-fields-to-skip

**Comment by ZhangMan1126** [comment]
@arcuri82 above comments have been addressed. Since one of your comments is related to logging, I then merge another branch into this one. That branch fixes some problems identified in Meituan's case studies and collects error information which exists in their specified seeded tests.

**Pull request merged**